app/code/Magento/Sales/Model/Order/Email/Sender/CreditmemoSender.php

```diff
@@ -100,12 +100,11 @@ public function __construct(
      */
     public function send(Creditmemo $creditmemo, $forceSyncMode = false)
     {
+        $this->identityContainer->setStore($creditmemo->getStore());
```

Comment:
It feels more like a workaround as it's changing the shared object that was injected through DI.

Could you provide info on where it used? Can we create a new identity container using the factory, set the correct store, and pass it further where it used?

Reply:
@ihor-sviziev 
It's not injected via di, CreditemmoSender extends from abstract Sender which has dependency on identity interface. Here we use creditmemo identity container implementation for this interface. As far as I found, the creditmemo identity container not used in other places.
As you may see, such workaround has been used from the beginning(and it's true for almost all senders, including abstract one), set store just been called to late which led to issues.
Yes, it'll be more cleaner to create creditmemo identity container via factory, but we already have coupling between objects to high for this sender and adding new dependency will make it even worse.

Reply:
Ok. I see there used `shared="false"` in order to prevent state changing. Let's keep it as is for now.

https://github.com/magento/magento2/blob/91aaeec6b2ea79f364b5614ac931972d00c085b0/app/code/Magento/Sales/etc/di.xml#L334

Note: seems like it used only here https://github.com/magento/magento2/blob/c4570d2efd58e4a2dd062994450c1e3decddd1b0/app/code/Magento/Sales/Model/Order/Email/Sender.php#L144-L150